tests/testthat/test-cast-list.R

```diff
@@ -15,3 +15,10 @@ test_that("elements of length >1 are truncated with warning", {
   out <- expect_warning(vec_list_cast(x, double()), class = "warn_lossy_cast")
   expect_equal(out, c(1, 2, 3))
 })
+
+test_that("error if all casts are lossy", {
+  # Do we really want to err out here? Perhaps give a different warning?
```

Comment:
Can you file an issue? It's possible we need a useful friendly form and a stricter programming form.